cmd/lakectl/cmd/annotate.go

```diff
@@ -0,0 +1,53 @@
+package cmd
+
+import (
+	"github.com/spf13/cobra"
+	"github.com/treeverse/lakefs/pkg/api"
+)
+
+const annotateTemplate = `{{ range $val := .Commits}}
+{{ $.Object|ljust 10}} {{ $val.Committer }} {{ $val.Id | printf "%.16s"|ljust 20 }} {{ $val.Message |ljust 30 }} {{ $val.CreationDate|date }}{{ end -}}
+`
+
+var annotateCmd = &cobra.Command{
+	Use:   "annotate <path uri>",
+	Short: "Show who created the latest commit",
+	Long:  "Show who created the latest commit to an object or objects in a give path",
+	Args:  cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, c_args []string) {
+		pathURI := MustParsePathURI("path", c_args[0])
+		client := getClient()
+		amount := 1
+		pfx := api.PaginationPrefix(*pathURI.Path)
+		res, err := client.ListObjectsWithResponse(cmd.Context(), pathURI.Repository, pathURI.Ref, &api.ListObjectsParams{Prefix: &pfx})
+		DieOnResponseError(res, err)
+		objList := res.JSON200.Results
+		logCommitsParams := &api.LogCommitsParams{
+			Amount: api.PaginationAmountPtr(amount),
+		}
+		pagination := api.Pagination{HasMore: true}
+		for _, obj := range objList {
```

Comment:
as far as I remember aws s3 and lakectl object listing continue to list all the object until no mo data is sent or a limit is specified.

Reply:
this command should be at least consistent with `lakectl fs ls` in behavior. If we do pagination there, we should probably do it here as well.

Reply:
Agree. Will be fixed.